crates/signer-ledger/src/app.rs

```diff
@@ -0,0 +1,347 @@
+            let offset = 1 + result[0] as usize;
+            let address_str = &result[offset + 1..offset + 1 + result[offset] as usize];
+            let mut address = [0; 20];
+            address.copy_from_slice(&hex::decode(address_str)?);
+            Address::from(address)
+        };
+        debug!(?address, "Received address from device");
+        Ok(address)
+    }
+
+    /// Returns the semver of the Ethereum ledger app
+    pub async fn version(&self) -> Result<String, LedgerError> {
+        let transport = self.transport.lock().await;
+
+        let command = APDUCommand {
+            ins: INS::GET_APP_CONFIGURATION as u8,
+            p1: P1::NON_CONFIRM as u8,
+            p2: P2::NO_CHAINCODE as u8,
+            data: APDUData::new(&[]),
+            response_len: None,
+        };
+
+        debug!("Dispatching get_version");
+        let answer = block_on(transport.exchange(&command))?;
+        let result = answer.data().ok_or(LedgerError::UnexpectedNullResponse)?;
+        if result.len() < 4 {
+            return Err(LedgerError::ShortResponse { got: result.len(), at_least: 4 });
+        }
+        let version = format!("{}.{}.{}", result[1], result[2], result[3]);
+        debug!(version, "Retrieved version from device");
+        Ok(version)
+    }
+
+    /// Signs an Ethereum transaction (requires confirmation on the ledger)
+    #[cfg(TODO)]
+    pub async fn sign_tx(&self, tx: &TypedTransaction) -> Result<Signature, LedgerError> {
+        let mut tx_with_chain = tx.clone();
+        if tx_with_chain.chain_id().is_none() {
+            // in the case we don't have a chain_id, let's use the signer chain id instead
+            tx_with_chain.set_chain_id(self.chain_id);
+        }
+        let mut payload = Self::path_to_bytes(&self.derivation);
+        payload.extend_from_slice(tx_with_chain.rlp().as_ref());
+
+        let mut signature = self.sign_payload(INS::SIGN, &payload).await?;
+
+        // modify `v` value of signature to match EIP-155 for chains with large chain ID
+        // The logic is derived from Ledger's library
+        // https://github.com/LedgerHQ/ledgerjs/blob/e78aac4327e78301b82ba58d63a72476ecb842fc/packages/hw-app-eth/src/Eth.ts#L300
+        let eip155_chain_id = self.chain_id * 2 + 35;
+        if eip155_chain_id + 1 > 255 {
+            let one_byte_chain_id = eip155_chain_id % 256;
+            let ecc_parity = if signature.v > one_byte_chain_id {
+                signature.v - one_byte_chain_id
+            } else {
+                one_byte_chain_id - signature.v
+            };
+
+            signature.v = match tx {
+                TypedTransaction::Eip2930(_) | TypedTransaction::Eip1559(_) => {
+                    (ecc_parity % 2 != 1) as u64
+                }
+                TypedTransaction::Legacy(_) => eip155_chain_id + ecc_parity,
+                #[cfg(feature = "optimism")]
+                TypedTransaction::DepositTransaction(_) => 0,
```

Comment:
Deposit transactions never contain signatures since they are side effects of events emitted from L1. It should never hit this branch so perhaps should `panic` or error if a user tries to sign a `DepositTransaction`?

Reply:
1. ported 1-1 from ethers
2. we don't even have this type so it's configured out for now

Reply:
nothing in this workspace should have optimism support yet either. please remove